File: src/main/scala/committee/nova/protonvp/event/ForgeEventHandler.scala
package committee.nova.protonvp.event

import committee.nova.proton.core.event.impl.ProtonPermNodeInitializationEvent
import committee.nova.protonvp.perm.PermissionList
import cpw.mods.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.common.MinecraftForge

import scala.collection.JavaConversions._

object ForgeEventHandler {
  def init(): Unit = MinecraftForge.EVENT_BUS.register(new ForgeEventHandler)
}

class ForgeEventHandler {
  @SubscribeEvent
  def onPermInit(event: ProtonPermNodeInitializationEvent): Unit = for (n <- PermissionList.VANILLA) event.addNodeFromString(s"minecraft.cmd.$n")
}

File: src/main/java/committee/nova/pvperms/event/ForgeEventHandler.java
package committee.nova.pvperms.event;

import committee.nova.proton.core.event.impl.ProtonPermNodeInitializationEvent;
import committee.nova.pvperms.ProtonVanillaPermissions;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import scala.collection.JavaConverters;

@Mod.EventBusSubscriber
public class ForgeEventHandler {
    @SubscribeEvent
    public static void onPermInit(ProtonPermNodeInitializationEvent event) {
        event.addNodeFromString(JavaConverters.asScalaIteratorConverter(ProtonVanillaPermissions.getPermsPrefixed().iterator()).asScala().toSeq());
    }
}
